contributing/BACKENDS.md

```diff
@@ -1,96 +1,165 @@
+5. **Data Quality Tests for Offline Providers**:
+- If your provider is classified as `offline`, you should add data quality tests to ensure the integrity of the precomputed CSV files. These tests are located in:
+  ```
+  src/integrity_tests/test_<yourprovider>.py
+  ```
+- Replace `<yourprovider>` with the name of your cloud provider. These tests verify the generated CSV files before publication to ensure accuracy and reliability.
 
-Create configuration models in `src/dstack/_internal/core/models/backends/<yourprovider>.py`. `<YourProvider>ConfigInfo`
-contains everything except for the credentials. You may have multiple models for credentials (i.e., default
-credentials & explicit credentials). Create a model with creds: `<YourProvider>ConfigInfoWithCreds`. Create a model with
-all fields being optional: `<YourProvider>ConfigInfoWithCredsPartial`. Create a model representing UI elements for
-configurator: `<YourProvider>ConfigValues`.
 
-Import all created models to `src/dstack/_internal/core/models/backends/__init__.py`.
+## Integrating a Cloud Provider into dstackai/dstack
 
-Implement `<YourProvider>Configurator`
-in `src/dstack/_internal/server/services/backends/configurators/<yourprovider>.py`
+Integrating a new cloud provider into `dstack` involves several key steps, from setting up your development environment to implementing specific backend configurations. Here’s how to proceed:
 
-Add `<YourProvider>Config` in `src/dstack/_internal/server/services/config.py`. This model represents the YAML
-configuration.
+### Setup and Initial Configuration
 
-Add safe import for your backend in `src/dstack/_internal/server/services/backends/__init__.py`. Update expected
-backends in tests in `src/tests/_internal/server/routers/test_backends.py`.
+1. **Clone the `dstack` Repository**: Begin by cloning the `dstack` repository from GitHub:
 
-## Appendix
+```bash
+git clone https://github.com/dstackai/dstack.git
+```
+
+2. **Follow Setup Instructions**: Consult the `CONTRIBUTING.md` document within the repository for instructions on setting up your development environment.
+
+### Modifying `setup.py`
+
+1. **Add Dependencies**: Incorporate any dependencies required by your cloud provider into `setup.py`. Create a separate section named `<yourprovider>` for these dependencies and ensure to update the `all` section to include them.
+
+### Extending Backend Models
+
+1. **Add Backend Type**: Insert a new enumeration entry for your backend in `src/dstack/_internal/core/models/backends/base.py`:
+
+```python
+<YOURBACKEND> = '<your_backend>'
+```
+2. **Create Provider Directory**: Establish a new directory at `src/dstack/_internal/core/backends/<yourprovider> `to house your provider’s backend and compute implementations.
+
+
+3. **Backend Implementation:** 
+In `__init__.py`, implement `YourProviderBackend`, inheriting from `BaseBackend`. Define the `TYPE` class variable to associate your backend with the newly added enum entry.
```

Comment:
Placeholders are not consistent. Use only `YourProvider` or `<YourProvider>` everywhere, not both.